flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -275,8 +275,28 @@
 	 *
 	 * @param table The table for which the AST and execution plan will be returned.
 	 */
+	@Deprecated
 	String explain(Table table);
 
+	/**
+	 * Returns the AST of the specified Table API and SQL queries and the execution plan to compute
+	 * the result of the given {@link Table}.
+	 *
+	 * @param table The table for which the AST and execution plan will be returned.
+	 * @param extended if the plan should contain additional properties such as
+	 * e.g. estimated cost, traits
+	 */
+	String explain(Table table, boolean extended);
+
+	/**
+	 * Returns the AST of the specified Table API and SQL queries and the execution plan to compute
+	 * the result of multiple-sinks plan.
+	 *
+	 * @param extended if the plan should contain additional properties such as
+	 * e.g. estimated cost, traits
+	 */
+	String explain(boolean extended);
```

Comment:
Let's add this in the second PR. We do not have the concept of lazy evaluation yet.

Reply:
The main goal of this PR is to introduce `explain(Boolean)` method, otherwise this PR is meaningless.

Reply:
Let's rebase it on top of #8984 then